include/arch/arm/arch.h

```diff
@@ -43,9 +43,11 @@ extern "C" {
 #include <arch/arm/cortex_m/nmi.h>
 #endif
 #if defined(CONFIG_MPU_STACK_GUARD)
-#define STACK_ALIGN  32
+#define STACK_ALIGN		32
+#define STACK_GUARD_SIZE	4
```

Comment:
Can we add a comment about what STACK_GUARD_SIZE is.

I assume this is carving out 4 bytes for the "poison" mpu region